File: painel_aplicativos/lib/app_service.dart
/// Serviço de dados (API simulação).
/// Centraliza busca e gravação em memória, espelhando a tabela do banco.
class AppService {
  /// Lista em memória simulando as linhas da tabela do banco de dados.
  static List<Map<String, dynamic>> tabelaAplicativos = [
    {
      'nome': 'MeuAppExemplo',
      'plataforma': 'Android Nativo',
      'versao': '1.0.0',
      'componentes': [
        {'tipo': 'Activity', 'nome': 'MainActivity'},
      ],
    },
  ];

  /// Retorna todos os aplicativos (SELECT *).
  List<Map<String, dynamic>> buscarTodos() {
    return List<Map<String, dynamic>>.from(tabelaAplicativos);
  }

  /// Insere um novo aplicativo (INSERT).
  void inserirNovo(Map<String, dynamic> novoApp) {
    tabelaAplicativos.add(Map<String, dynamic>.from(novoApp));
  }

  /// Remove por índice (DELETE) — base para exclusão por swipe.
  bool excluirPorIndice(int index) {
    if (index < 0 || index >= tabelaAplicativos.length) return false;
    tabelaAplicativos.removeAt(index);
    return true;
  }

  /// Filtra por nome (barra de pesquisa).
  List<Map<String, dynamic>> buscarPorNome(String termo) {
    final q = termo.trim().toLowerCase();
    if (q.isEmpty) return buscarTodos();
    return tabelaAplicativos
        .where((app) => (app['nome'] as String).toLowerCase().contains(q))
        .map((app) => Map<String, dynamic>.from(app))
        .toList();
  }
}

File: modules/app_service.py
"""Serviço de dados (API simulação) — equivalente Python de app_service.dart."""

from __future__ import annotations

from copy import deepcopy
from typing import Any


class AppService:
    """Centraliza busca e gravação em memória (simula tabela do banco)."""

    tabela_aplicativos: list[dict[str, Any]] = [
        {
            "nome": "MeuAppExemplo",
            "plataforma": "Android Nativo",
            "versao": "1.0.0",
            "componentes": [{"tipo": "Activity", "nome": "MainActivity"}],
        }
    ]

    def buscar_todos(self) -> list[dict[str, Any]]:
        return deepcopy(self.tabela_aplicativos)

    def inserir_novo(self, novo_app: dict[str, Any]) -> None:
        self.tabela_aplicativos.append(deepcopy(novo_app))

    def excluir_por_indice(self, index: int) -> bool:
        if index < 0 or index >= len(self.tabela_aplicativos):
            return False
        self.tabela_aplicativos.pop(index)
        return True

    def buscar_por_nome(self, termo: str) -> list[dict[str, Any]]:
        q = termo.strip().lower()
        if not q:
            return self.buscar_todos()
        return [
            deepcopy(app)
            for app in self.tabela_aplicativos
            if q in str(app.get("nome", "")).lower()
        ]


if __name__ == "__main__":
    svc = AppService()
    print(svc.buscar_todos())
